x-pack/plugins/observability_solution/apm/common/waterfall/typings.ts

```diff
@@ -8,81 +8,58 @@
 import { Exception } from '../../typings/es_schemas/raw/error_raw';
 import { EventOutcome } from '../../typings/es_schemas/raw/fields/event_outcome';
 import { SpanLink } from '../../typings/es_schemas/raw/fields/span_links';
-import { TimestampUs } from '../../typings/es_schemas/raw/fields/timestamp_us';
 import { AgentName } from '../../typings/es_schemas/ui/fields/agent';
 
 export interface WaterfallTransaction {
-  timestamp: TimestampUs;
-  trace: { id: string };
-  service: {
-    name: string;
-    environment?: string;
-  };
-  agent: { name: AgentName };
-  event?: { outcome?: EventOutcome };
-  parent?: { id?: string };
-  processor: { event: 'transaction' };
-  transaction: {
-    duration: { us: number };
-    id: string;
-    name: string;
-    type: string;
-    result?: string;
-  };
-  faas?: {
-    coldstart?: boolean;
-  };
-  span?: {
-    links?: SpanLink[];
-  };
+  'timestamp.us': number[];
+  'trace.id': string[];
+  'service.name': string[];
+  'service.environment'?: string[];
+  'agent.name': AgentName[];
+  'event.outcome'?: EventOutcome[];
+  'parent.id'?: string[];
+  'processor.event': ['transaction'];
+  'transaction.duration.us': number[];
+  'transaction.id': string[];
+  'transaction.name': string[];
+  'transaction.type': string[];
+  'transaction.result'?: string[];
+  'faas.coldstart': boolean[];
+  'span.links'?: SpanLink[][];
```

Comment:
We need to verify taking span links from `fields` works for our use case. My suspicion is that we need to take span links from `_source` because `fields` doesn't return an array of span link objects, but arrays for all leaf fields, such as `span.links.trace.id`. Also, the content for OTel span links and APM span links will look slightly differently.

Reply:
Yes, it appears both will have to be specified: `span.links.trace.id` and `span.links.span.id`. Are there any additional leaf fields? I can experiment with collecting them and dedotting.

Reply:
it seems like `span.links.trace.id` and `span.links.trace.id` are string types, while the processed structure is an array of `SpanLink[]` which implies there may be more than one span link per span, but I'm not able to find examples of spans with more than one `trace.id` and one `span.id`.

Reply:
I did some tests and it appears that span.links.trace.id and span.links.span.id will be two equal length and the ids at relative indices are associated. 
<img width="3120" alt="Screenshot 2024-09-05 at 14 31 15" src="https://github.com/user-attachments/assets/dbe5053b-9c33-4a14-ba32-a70626245517">

I was able to recreate the `SpanLink[]` structure by merging these two arrays together. A commit is coming shortly.